src/graphics.zig

```diff
@@ -2145,16 +2147,19 @@ pub const TextureArray = struct { // MARK: TextureArray
 
 	pub fn init() TextureArray {
 		var self: TextureArray = undefined;
-		c.glGenTextures(1, &self.textureID);
+		if(!main.settings.launchConfig.headlessServer)
+			c.glGenTextures(1, &self.textureID);
```

Comment:
I don't entirely love that I have to wrap a ton of stuff with these ifs, particularly because deinit and bind do absolutely nothing if headless is true. So... Maybe that could be better.